src/hooks/users/useRecentWorldUsers.ts

```diff
@@ -1,30 +1,44 @@
-import { useMemo } from "react";
+import { isLoaded } from "react-redux-firebase";
+import { isEqual } from "lodash";
 
-import { User } from "types/User";
+import { User, userWithLocationToUser } from "types/User";
 
 import { WithId } from "utils/id";
+import { worldUsersSelector } from "utils/selectors";
 import { normalizeTimestampToMilliseconds } from "utils/time";
 
+import { useSelector } from "hooks/useSelector";
+import { useConnectWorldUsers } from "hooks/users";
 import { useUserLastSeenThreshold } from "hooks/useUserLastSeenThreshold";
 
-import { useWorldUsers } from "./useWorldUsers";
+const noUsers: WithId<User>[] = [];
 
 export const useRecentWorldUsers = (): {
   recentWorldUsers: readonly WithId<User>[];
   isRecentWorldUsersLoaded: boolean;
 } => {
   const lastSeenThreshold = useUserLastSeenThreshold();
 
-  const { worldUsers, isWorldUsersLoaded } = useWorldUsers();
+  useConnectWorldUsers();
 
-  return useMemo(
-    () => ({
-      recentWorldUsers: worldUsers.filter(
+  const { recentWorldUsers, isWorldUsersLoaded } = useSelector((state) => {
+    const worldUsers = worldUsersSelector(state);
+    const isWorldUsersLoaded = isLoaded(worldUsers);
+
+    if (!worldUsers) return { recentWorldUsers: noUsers, isWorldUsersLoaded };
+
+    const recentWorldUsers = worldUsers
+      .filter(
```

Comment:
Same question as above. How do we have the users in the redux state, given that we don't fetch them

Reply:
See my answer in https://github.com/sparkletown/sparkle/pull/1703#discussion_r666010508, but I believe the context provider still uses redux.

Reply:
Marking this as resolved, reopen if needs be.